### PR TITLE
NTP authentication

### LINKED ISSUE #820: NTP authentication
**Issue Type**

- [ ] Template enhancement
- [x] Role enhancement
- [ ] Documentation enhancement (avd.sh)

**Is your feature request related to a problem? Please describe.**
We will need NTP authentication for security

```
ntp authentication-key 1 sha1 7 8493utgdd2
ntp authenticate
ntp trusted key 328
```
Can be added to the same structure as ntp.j2

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by anavazio**
## Change Summary

Added NTP authentication for one or more sha1 or md5 keys.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [x] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #820

## Component(s) name

`arista.avd.eos_cli_config-gen`

## Proposed changes
Add NTP authentication using either SHA1 or MD5 hashes, with the ability to add trusted and untrusted keys.

Defining "keys" within ntp_server activates NTP authentication.
An arbitrary id from 1 to 65534 inclusive is required.
A key is assumed to be trusted unless defined otherwise.

`keys:
    <id>:
      hash: <md5 or sha1 hashed password>
      hash_algorithm: < md5 | sha1 >
      trusted: < true | false > (optional)`

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

**Review by carlbuchmann — CHANGES_REQUESTED** [review]
@anavazio thanks for submitting this PR!

Need to adjust data model to follow our standards for eos_cli_config_gen which provides no abstraction.

```yaml
ntp_authenticate: <true | false >

ntp_authentication_keys:
  <key_identifier | 1-65534>:
    hash_algorithm: < md5 | sha1 >
    key:  "< encypted_key >"

ntp_trusted_keys: "< list of trusted-keys as string >"
```
@onurgashi can you please review and commend if you agree?

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/ntp.md [hunk 1]
```diff
@@ -60,13 +60,22 @@ interface Management1
 | 10.1.1.1 | true |
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/ntp.md:64]: you submitted generated artifacts, but the PR doesn't include the augmented test case -> https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/ntp.yml

**Review by titom73 — COMMENTED** [review]
I can't see the generated configuration file in molecule. So I expect CI to fail as it will generate diffs with current version.

Could you also run molecule and update artifacts in this PR ?

Also, some trailing whitespaces are present in your README section. So you should remove them to pass the CI.

**Comment by anavazio** [comment]
@titom73 I'm currently having an issue installing the container for development, so I'll get on that once I have it installed. Thanks.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/ntp.md [hunk 2]
```diff
@@ -60,13 +60,22 @@ interface Management1
...
+|  10  |    356708192346f678a878d6e99889767a    |  md5 |   True  |
+
 ### NTP Device Configuration
 
 ```eos
 !
 ntp local-interface vrf test lo1
 ntp server vrf test 10.1.1.1 prefer
 ntp server vrf test 10.1.1.2
+
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/ntp.md]: This is more cosmetic, but try to render the CLI output match as close as possible to the order on EOS device. Also, remove the new line introduced.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/ntp.yml [hunk 3]
```diff
@@ -5,4 +5,13 @@ ntp_server:
...
\ No newline at end of file
+    - 10.1.1.2
+  ntp_authenticate: true
+
+  ntp_authentication_keys:
+    10:
+      hash_algorithm: "md5"
+      key:  "356708192346f678a878d6e99889767a"
+  
+  ntp_trusted_keys:
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/ntp.yml]: `ntp_trusted_keys` should be a simple string containing multiple keys and not a list. This is similar to VLANs on interfaces... perhaps we need to make the documentation more clear?

example of possible values:

```yaml
 ntp_trusted_keys: "10"
 ntp_trusted_keys: "10-12,15"
 ntp_trusted_keys: "10,12,15"
```

**Comment by ClausHolbechArista** [comment]
@anavazio We will release 2.2 very soon, so to get this PR merged in time, we need to get it finished today or tomorrow. I can help get it over the finish line if it is okay with you. I would submit fixes directly to your branch. Please let me know.

**Comment by anavazio** [comment]
@ClausHolbechArista Of course! Thank you!

**Review by carlbuchmann — APPROVED** [review]
LGMT!

**Pull request merged**